watch.go

```diff
@@ -21,7 +23,23 @@ type Event struct {
 	// Path to the data that changed
 	Path string
 	// Data that changed
-	Data interface{}
+	Data    interface{}
+	payload string
+}
+
+func (e Event) Value(v interface{}) (path string, err error) {
```

Comment:
This is up to you. If you want to keep the "old style" forever, then I
agree that it is OK to return the error only. But if you plan to make the
system more efficient by deprecating the "old style", then you will need a
way to get the path and you will probably have to then change the API.

So I suggest that you keep the path in the return values, but it is your
package and you can decide what you want. Just let me know your final
decision and I will code accordingly.

█  Jacques

On 2 August 2016 at 04:29, Steven Berlanga <notifications@github.com> wrote:

> In watch.go
> <https://github.com/zabawaba99/firego/pull/50#discussion_r73085472>:
>
> > @@ -21,7 +23,23 @@ type Event struct {
> >  	// Path to the data that changed
> >  	Path string
> >  	// Data that changed
> > -	Data interface{}
> > +	Data    interface{}
> > +	payload string
> > +}
> > +
> > +func (e Event) Value(v interface{}) (path string, err error) {
>
> In order for us to see those gains though, we would have to remove the old
> style. For the time being, I think we should return an error from this
> function and just have the consumer access the path through the Event
> struct.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zabawaba99/firego/pull/50/files/91637d68c23bcd6761930b23882b18621b759365#r73085472>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAB2iG6KYx7CzW6h4pMTrynlV2PPcjjnks5qbqt6gaJpZM4JYHTS>
> .
>